src/atomate2/abinit/jobs/core.py

```diff
@@ -0,0 +1,178 @@
+
+
+@dataclass
+class NonSCFMaker(BaseAbinitMaker):
+    """Maker to create non SCF calculations."""
+
+    calc_type: str = "nscf"
+    name: str = "non-Scf calculation"
+
+    input_set_generator: AbinitInputGenerator = field(
+        default_factory=NonSCFSetGenerator
+    )
+
+    # Non dataclass variables:
+    CRITICAL_EVENTS: ClassVar[Sequence[str]] = ("NscfConvergenceWarning",)
+
+    @job
+    def make(
+        self,
+        structure: Structure | None = None,
+        prev_outputs: str | list[str] | None = None,
+        restart_from: str | list[str] | None = None,
+        history: JobHistory | None = None,
+        mode: str = "uniform",
+    ):
+        """
+        Run a non-scf ABINIT job.
+
+        Parameters
+        ----------
+        structure : .Structure
+            A pymatgen structure object.
+        mode : str
+            Type of band structure calculation. Options are:
+            - "line": Full band structure along symmetry lines.
+            - "uniform": Uniform mesh band structure.
+        """
+        self.input_set_generator.mode = mode
+
+        return super().make.original(
+            self,
+            structure=structure,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            history=history,
+        )
+
+
+@dataclass
+class NonSCFWfqMaker(NonSCFMaker):
+    """Maker to create non SCF calculations for the WFQ."""
+
+    calc_type: str = "nscf_wfq"
+    name: str = "non-Scf calculation"
+
+    input_set_generator: AbinitInputGenerator = field(
+        default_factory=NonScfWfqInputGenerator
+    )
+
+    # Non dataclass variables:
+    CRITICAL_EVENTS: ClassVar[Sequence[str]] = ("NscfConvergenceWarning",)
+
+
+@dataclass
+class RelaxMaker(BaseAbinitMaker):
+    """Maker to create relaxation calculations."""
+
+    calc_type: str = "relax"
+    input_set_generator: AbinitInputGenerator = field(default_factory=RelaxSetGenerator)
+    name: str = "Relaxation calculation"
+
+    # non-dataclass variables
+    CRITICAL_EVENTS: ClassVar[Sequence[str]] = ("RelaxConvergenceWarning",)
+    # structure_fixed: ClassVar[bool] = False
+
+    @classmethod
+    def ionic_relaxation(cls, *args, **kwargs):
+        """Create an ionic relaxation maker."""
+        # TODO: add the possibility to tune the RelaxInputGenerator options
+        #  in this class method.
+        return cls(
+            # input_set_generator=RelaxSetGenerator(relax_cell=False, *args, **kwargs),
+            input_set_generator=RelaxSetGenerator(relax_cell=False, **kwargs),
+            name=cls.name + " (ions only)",
+        )
+
+    @classmethod
+    def full_relaxation(cls, *args, **kwargs):
+        """Create a full relaxation maker."""
+        # TODO: add the possibility to tune the RelaxInputGenerator options
+        #  in this class method.
+        return cls(
+            # input_set_generator=RelaxSetGenerator(relax_cell=True, *args, **kwargs)
```

Comment:
This line seems like it should provide full control. Any reason it is commented out?